hasher-matcher-actioner/hmalib/common/models/tests/test_bank_member_with_key.py

```diff
@@ -14,41 +17,100 @@
 
 class BankMemberWithKeyTestCase(BanksTableTestBase, unittest.TestCase):
     def _create_bank(self) -> str:
+        """Create a bank with a unique id."""
+        if not hasattr(self, "_id_count"):
+            self._id_count = 0
+
+        self._id_count = self._id_count + 1
+
         table_manager = BanksTable(
             self.get_table(), signal_type_mapping=get_default_signal_type_mapping()
         )
 
-        bank = table_manager.create_bank("TEST_Bank", "Test bank description")
+        bank = table_manager.create_bank(
+            f"TEST_Bank_{self._id_count}", "Test bank description"
+        )
         return bank.bank_id
 
+    def _add_photo_bank_member_with_key(
+        self, table_manager: BanksTable, bank_id: str, member_key: str
+    ):
+        return table_manager.add_keyed_bank_member(
+            bank_id=bank_id,
+            member_key=member_key,
+            content_type=PhotoContent,
+            storage_bucket=None,
+            storage_key=None,
+            raw_content="does not matter",
+            notes="",
+            is_media_unavailable=True,
+        )
+
     def test_add_bank_member_with_key(self):
-        with self.fresh_dynamodb():
-            table_manager = BanksTable(
-                self.get_table(), get_default_signal_type_mapping()
+        with self.fresh_table_manager() as table_manager:
+            bank_id = self._create_bank()
+            member_key = "fooofofofofoof"
+
+            self._add_photo_bank_member_with_key(table_manager, bank_id, member_key)
+
+            with self.assertRaises(KeyError):
+                self._add_photo_bank_member_with_key(table_manager, bank_id, member_key)
+
+    def test_add_keyed_bank_member_in_multiple_banks(self):
+        with self.fresh_table_manager() as table_manager:
+            bank_id = self._create_bank()
+            bank_id_2 = self._create_bank()
+
+            member_key = "a-very-unique-id-as-you-can-see"
+
+            bank_member = self._add_photo_bank_member_with_key(
+                table_manager, bank_id, member_key
             )
 
+            bank_member_2 = self._add_photo_bank_member_with_key(
+                table_manager, bank_id_2, member_key
+            )
+
+            self.assertNotEqual(bank_member.bank_member_id, bank_member_2)
+
+    def test_add_multiple_signals_for_keyed_bank_member(self):
```

Comment:
Sure. I can whip something up.

Reply:
This is proving to be harder than I thought. I thought the NCMEC API would be best for this test because the data model atleast allows me to return more than one signal type per row..

However, `[_get_conversion](https://github.com/facebook/ThreatExchange/blob/main/python-threatexchange/threatexchange/exchanges/impl/ncmec_api.py#L115)` prevents me from doing exactly that. 

I'd have to invent a new SignalExchangeAPI to get this to work. Back-burnering this ask as #1219 instead.